src/Faker/Caching/ResourceCollectionCacher.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+
+namespace Faker.Caching
+{
+	/// <summary>
+	///   This class caches resource items containing collections of strings, separated by a separator.
+	/// </summary>
+	/// <remarks>
+	///   It returns the splitted array reducing complexity from O(n) to O(1). The class is thread-safe.
+	/// </remarks>
+	internal static class ResourceCollectionCacher
+	{
+		private static object lock_obj = new object();
+		private static Dictionary<string, string[]> cache = new Dictionary<string, string[]>();
+
+		internal static string[] GetArray(PropertyInfo p)
+		{
+			var invokingClassName = p.DeclaringType.FullName;
+			var invokedPropertyName = p.Name;
+			var currentUICultureName = global::System.Globalization.CultureInfo.CurrentUICulture.Name;
```

Comment:
You don't need to use `global::`
And since already imported `using System.Globalization` I would prefer you changed
this to `CultureInfo.CurrentUICulture.Name`